internal/pkg/action/operator_uninstall.go

```diff
@@ -74,41 +66,68 @@ func (u *OperatorUninstall) Run(ctx context.Context) error {
 		return fmt.Errorf("get subscription CSV %q: %v", csvName, err)
 	}
 
+	// Subscriptions can be deleted asynchronously.
+	if err := u.deleteObjects(ctx, sub); err != nil {
+		return err
+	}
+
+	// If we could not find a csv associated with the subscription, that likely
+	// means there is no CSV associated with it yet. Return a nil error.
+	if csv == nil {
+		u.Logf("csv for package %q not found", u.Package)
+		return nil
+	}
+
 	// Deletion order:
 	//
 	// 1. Subscription to prevent further installs or upgrades of the operator while cleaning up.
-	// 2. CustomResourceDefinitions so the operator has a chance to handle CRs that have finalizers.
+	// 2. Operands so the operator has a chance to handle CRs that have finalizers.
+	// Note: the correct strategy must be chosen in order to process an opertor delete with operand on-cluster.
 	// 3. ClusterServiceVersion. OLM puts an ownerref on every namespaced resource to the CSV,
 	//    and an owner label on every cluster scoped resource so they get gc'd on deletion.
 
-	// Subscriptions can be deleted asynchronously.
-	if err := u.deleteObjects(ctx, sub); err != nil {
-		return err
+	// create lister and list operands
+	lister := action.NewOperatorListOperands(u.config)
+	operands, err := lister.Run(ctx, u.Package)
+	if err != nil {
+		return fmt.Errorf("listing operands associated with operator %s: %s", u.Package, err)
 	}
 
-	if csv != nil {
-		// Ensure CustomResourceDefinitions are deleted next, so that the operator
-		// has a chance to handle CRs that have finalizers.
-		if u.DeleteCRDs {
-			crds := getCRDs(csv)
-			if err := u.deleteObjects(ctx, crds...); err != nil {
+	switch u.OperandStrategy.Kind {
+	case operand.Cancel:
+		if len(operands.Items) > 0 {
+			return fmt.Errorf("%d operands exist and operand strategy %q is in use; "+
+				"delete operands or re-run with a different operand strategy", len(operands.Items), operand.Cancel)
+		}
+	case operand.Ignore:
+		for _, op := range operands.Items {
+			u.Logf("%s %q orphaned", strings.ToLower(op.GetKind()), op.GetName()+"/"+op.GetNamespace())
```

Comment:
Some operands will be cluster-scoped and therefore not have namespaces. We should handle that case separately instead of printing something like `clusterissuer "my-cluster-issuer/" orphaned`.

Perhaps there a helper type in a k8s library that can handle this for us?